homeassistant/components/music_assistant/media_player.py

```diff
@@ -0,0 +1,729 @@
+            enqueue=enqueue,
+            media_type=media_type,
+            radio_mode=kwargs[ATTR_MEDIA_EXTRA].get(ATTR_RADIO_MODE),
+        )
+
+    @catch_musicassistant_error
+    async def async_join_players(self, group_members: list[str]) -> None:
+        """Join `group_members` as a player group with the current player."""
+        player_ids: list[str] = []
+        for child_entity_id in group_members:
+            # resolve HA entity_id to MA player_id
+            if (hass_state := self.hass.states.get(child_entity_id)) is None:
+                continue
+            if (mass_player_id := hass_state.attributes.get("mass_player_id")) is None:
+                continue
+            player_ids.append(mass_player_id)
+        await self.mass.players.cmd_sync_many(self.player_id, player_ids)
+
+    @catch_musicassistant_error
+    async def async_unjoin_player(self) -> None:
+        """Remove this player from any group."""
+        await self.mass.players.player_command_unsync(self.player_id)
+
+    @catch_musicassistant_error
+    async def _async_play_media_advanced(
+        self,
+        media_id: list[str],
+        artist: str | None = None,
+        album: str | None = None,
+        enqueue: MediaPlayerEnqueue | QueueOption | None = None,
+        radio_mode: bool | None = None,
+        media_type: str | None = None,
+    ) -> None:
+        """Send the play_media command to the media player."""
+        # pylint: disable=too-many-arguments
+        media_uris: list[str] = []
+        item: MediaItemType | None = None
+        # work out (all) uri(s) to play
+        for media_id_str in media_id:
+            # URL or URI string
+            if "://" in media_id_str:
+                media_uris.append(media_id_str)
+                continue
+            # try content id as library id
+            if media_type and media_id_str.isnumeric():
+                with suppress(MediaNotFoundError):
+                    if media_type is not None:
+                        continue
+                    item = await self.mass.music.get_item(  # type: ignore[unreachable]
+                        MediaType(media_type), media_id_str, "library"
+                    )
+                    if item.uri is not None:
+                        continue
+                    media_uris.append(item.uri)
+                    continue
+            # try local accessible filename
+            elif await asyncio.to_thread(os.path.isfile, media_id_str):
+                media_uris.append(media_id_str)
+                continue
+            # last resort: lookup by name/search
+            if item := await self._get_item_by_name(
+                media_id_str, artist, album, media_type
+            ):
+                if TYPE_CHECKING:
+                    assert item.uri is not None
+                media_uris.append(item.uri)
+
+        if not media_uris:
+            raise MediaNotFoundError(
```

Comment:
We should not raise library exceptions

Reply:
The error should be caught by the exception handling wrapper and converted to a HomeAssistantError

Reply:
I think we should raise HomeAssistantError here